src/plugins/hetero/src/config.cpp

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "config.hpp"
+
+// #include <cpp_interfaces/interface/ie_internal_plugin_config.hpp>
+#include <ie_plugin_config.hpp> // TODO vurusovs create file with custom properties
+
+#include "openvino/runtime/internal_properties.hpp"
+#include "openvino/runtime/properties.hpp"
+
+using namespace ov::hetero;
+
+Configuration::Configuration() {}
+
+// TODO vurusovs DELETE `throwOnUnsupported` option
+Configuration::Configuration(const ov::AnyMap& config, const Configuration& defaultCfg, bool throwOnUnsupported) {
+    *this = defaultCfg;
+
+    for (auto&& c : config) {
+        const auto& key = c.first;
+        const auto& value = c.second;
+
+        if (HETERO_CONFIG_KEY(DUMP_GRAPH_DOT) == key)   // TODO vurusovs: rewrite HETERO_CONFIG_KEY(DUMP_GRAPH_DOT)
+            dump_graph = value.as<bool>();
+        else if (ov::device::priorities.name() == key)
+            device_priorities = value.as<std::string>();
+        else if (ov::exclusive_async_requests == key) {
+            exclusive_async_requests = value.as<bool>();
+            m_device_config[key] = value;
+        }
+        else
+            m_device_config[key] = value;   // save property as is to pass to underlying device
```

Comment:
We should not get device properties here. It is `HETERO` config. Exception is needed.

Reply:
Updated - only HETERO properties (and `EXCLUSIVE_ASYNC_REQUESTS` as device property stored in HETERO) saved. In case of `set_property()` any unexpected property will cause exception. In any other cases (`Plugin()` or `HeteroExecutableNetwork()`/`CompiledModel()`) it won't cause exception

Reply:
I decided to rewrite logic because `exclusive_async_requests` property looks very strange - it's a part of hetero properties, but should be passed to device. And it isn't actually used. So I decided to store all properties in `Configuration` and split it on `HeteroProperties` and `DeviceProperties` parts